packages/graphql/src/translate/create-projection-and-params.ts

```diff
@@ -639,3 +621,31 @@ function createProjectionAndParams({
 }
 
 export default createProjectionAndParams;
+
+// Fields of reference node to sort on. Since sorting is done on projection, if field is not selected or is and aliased
+// sort will fail silently
+const generateSortFields = ({
+    node,
+    resolveTree,
+}: {
+    node: Node;
+    resolveTree: ResolveTree;
+}): Record<string, ResolveTree> => {
+    const nodeFields = resolveTree.fieldsByTypeName[node.name];
+
+    const sortFieldNames = ((resolveTree.args.options as GraphQLOptionsArg)?.sort ?? []).map(Object.keys).flat();
+
+    return Array.from(new Set(sortFieldNames)).reduce(
+        (acc, sortFieldName) => ({
+            ...acc,
+            // If fieldname is not found in fields of selection set
+            ...(!Object.values(nodeFields).find((field) => field.name === sortFieldName) ||
+            // or does exist but is aliased
+            Object.values(nodeFields).find((field) => field.name === sortFieldName && field.alias !== sortFieldName)
+                ? // generate a basic resolve tree
+                  generateProjectionField({ name: sortFieldName })
+                : {}),
+        }),
+        {}
+    );
```

Comment:
**suggestion:** This reduce statement is starting to rack up some complexity. I wonder if we could improve readability a bit by splitting it into multiple statements inside the function, something akin to:

```suggestion
    return Array.from(new Set(sortFieldNames)).reduce((acc, sortFieldName) => {
        const fieldMissing = !Object.values(nodeFields).find((field) => field.name === sortFieldName);
        const fieldIsAliased = Boolean(
            Object.values(nodeFields).find((field) => field.name === sortFieldName && field.alias !== sortFieldName)
        );

        if (fieldMissing || fieldIsAliased) {
            return {
                ...acc,
                ...generateProjectionField({ name: sortFieldName }),
            };
        }
        return acc;
    }, {} as Record<string, ResolveTree>);
```

Thoughts @dmoree ?

Reply:
Thanks for the suggestion. I do agree that this reads better.

Refactored in https://github.com/neo4j/graphql/pull/857/commits/96040d368a877e7096e9f7a0f45ad52f897d3466